### DISCUSSION TITLE
How to handle Refill and Bucket instances over multiple PODs?

### DISCUSSION THREAD

**Discussion by lfaltran** [vladimir-bukhtoyarov/bucket4j#272] [Q&A] [answered]
I've an application (spring boot) running over multiple pods on kubernetes.
Pod A, B, C...

All the instances consume remote services that have throttling management. Sometimes, Pod A consume the service, another Pod B and so on, on some circumstances, all Pods can try to access these remote services at same time, and my application has to handle these.

Working with "bucket.tryConsume" everything is running well, my problem still with "Refill" strategy, and how to share my Bandwidth across all Pods.

Am I missing something?

**Comment by vladimir-bukhtoyarov** [comment, marked as answer]
Hello @lfaltran 

You need to use [distributed buckets](https://bucket4j.com/7.5.0/toc.html#distributed-facilities). Bucket4j supports many backends for such case:
* https://github.com/bucket4j/bucket4j#supported-jcache-compatibleor-similar-back-ends
* https://github.com/bucket4j/bucket4j#non-jvm-back-ends

**Reply by lfaltran** [reply]
Thank you @vladimir-bukhtoyarov now I got it how to go ahead on my app
